lib/rack/timeout/core.rb

```diff
@@ -86,6 +101,11 @@ def initialize(app, service_timeout:nil, wait_timeout:nil, wait_overtime:nil, se
       @app = app
     end
 
+    # Overriding attribute readers to evaluate Procs
+    def service_timeout; proc_or_value(@service_timeout, @env); end
+    def wait_timeout; proc_or_value(@wait_timeout, @env); end
+    def wait_overtime; proc_or_value(@wait_overtime, @env); end
+    def service_past_wait; proc_or_value(@service_past_wait, @env); end
```

Comment:
If we're defining these methods here, then we should remove the `attr_reader`s of the same names above while keeping their documentation comments.

We also need to store `@env` to pass it into the Procs (or find a better pattern for sending the Rack env to attribute Procs)